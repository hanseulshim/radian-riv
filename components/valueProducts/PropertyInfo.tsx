import React from 'react'

export default function PropertyInfo() {
  return (
    <div className="property-info">
      <div className="property-info-header">
        <h1>RIV Property Info</h1>
        <button className="btn btn-primary lock-button">
          <span>Lock</span>
          <div className="lock-icon">
            <img src={`${process.env.baseUrl}/images/lock.svg`} alt="lock" />
          </div>
        </button>
      </div>
      <button style={{ marginRight: '1em' }} className="btn btn-primary">
        Re-Order RIV
      </button>
      <button style={{ marginRight: '1em' }} className="btn btn-primary">
        Order Rental Analysis
      </button>
      <button style={{ marginRight: '1em' }} className="btn btn-primary">
        Change Property Characteristics
      </button>
      <div className="icon-row">
        <img
          className="icon"
          src={`${process.env.baseUrl}/images/download-pdf.svg`}
          alt="logo"
        />
        <img
          className="icon"
          src={`${process.env.baseUrl}/images/download-excel.svg`}
          alt="download-excel"
        />
        <img
          className="icon"
          src={`${process.env.baseUrl}/images/aerial-map.svg`}
          alt="aerial-map"
        />
        <img
          className="icon"
          src={`${process.env.baseUrl}/images/cost.svg`}
          alt="cost"
        />
        <img
          className="icon"
          src={`${process.env.baseUrl}/images/property-info.svg`}
          alt="property-info"
        />
        <img
          className="icon"
          src={`${process.env.baseUrl}/images/photos.svg`}
          alt="photos"
        />
      </div>
    </div>
  )
}
