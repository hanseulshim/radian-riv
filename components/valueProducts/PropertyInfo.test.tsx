import * as React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

import PropertyInfo from './PropertyInfo'

describe('Property Info', () => {
  let component
  beforeAll(() => {
    component = shallow(<PropertyInfo />)
  })
  it('has h1 tag and lock button', () => {
    expect(component.find('h1').text()).toEqual('RIV Property Info')
    expect(component.find('.btn-icon').text()).toEqual('Lock')
    expect(component.find('.icon-container > img').prop('src')).toContain(
      'lock.svg'
    )
  })
  it('has three top row buttons', () => {
    expect(component.find('button').at(1).text()).toEqual('Re-Order RIV')
    expect(component.find('button').at(2).text()).toEqual(
      'Order Rental Analysis'
    )
    expect(component.find('button').at(3).text()).toEqual(
      'Change Property Characteristics'
    )
  })
  it('has 7 icons', () => {
    expect(component.find('.icon-row > .icon').at(0).prop('src')).toContain(
      'download-pdf.svg'
    )
    expect(component.find('.icon-row > .icon').at(1).prop('src')).toContain(
      'download-excel.svg'
    )
    expect(component.find('.icon-row > .icon').at(2).prop('src')).toContain(
      'aerial-map.svg'
    )
    expect(component.find('.icon-row > .icon').at(3).prop('src')).toContain(
      'cost.svg'
    )
    expect(component.find('.icon-row > .icon').at(4).prop('src')).toContain(
      'property-info.svg'
    )
    expect(component.find('.icon-row > .icon').at(5).prop('src')).toContain(
      'photos.svg'
    )
  })
  it('should match snapshot', () => {
    expect(toJson(component)).toMatchSnapshot()
  })
})
