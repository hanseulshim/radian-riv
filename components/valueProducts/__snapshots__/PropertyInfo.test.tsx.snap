// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Property Info should match snapshot 1`] = `
<div
  className="property-info"
>
  <div
    className="property-info-header"
  >
    <h1>
      RIV Property Info
    </h1>
    <button
      className="btn btn-primary lock-button"
    >
      <span>
        Lock
      </span>
      <div
        className="lock-icon"
      >
        <img
          alt="lock"
          src="undefined/images/lock.svg"
        />
      </div>
    </button>
  </div>
  <button
    className="btn btn-primary"
    style={
      Object {
        "marginRight": "1em",
      }
    }
  >
    Re-Order RIV
  </button>
  <button
    className="btn btn-primary"
    style={
      Object {
        "marginRight": "1em",
      }
    }
  >
    Order Rental Analysis
  </button>
  <button
    className="btn btn-primary"
    style={
      Object {
        "marginRight": "1em",
      }
    }
  >
    Change Property Characteristics
  </button>
  <div
    className="icon-row"
  >
    <img
      alt="logo"
      className="icon"
      src="undefined/images/download-pdf.svg"
    />
    <img
      alt="download-excel"
      className="icon"
      src="undefined/images/download-excel.svg"
    />
    <img
      alt="aerial-map"
      className="icon"
      src="undefined/images/aerial-map.svg"
    />
    <img
      alt="cost"
      className="icon"
      src="undefined/images/cost.svg"
    />
    <img
      alt="property-info"
      className="icon"
      src="undefined/images/property-info.svg"
    />
    <img
      alt="photos"
      className="icon"
      src="undefined/images/photos.svg"
    />
  </div>
</div>
`;
